cmd/collector/app/server/http_test.go

```diff
@@ -240,18 +248,21 @@ func TestSpanCollectorHTTPS(t *testing.T) {
 			} else {
 				require.NoError(t, requestError)
 				require.NotNil(t, response)
+				response.Body.Close()
```

Comment:
this should be on L245 as `defer ...`

Reply:
In a case when there is an error in the request, it throws a nil pointer since the request.Body might not have been initialized 

github.com/jaegertracing/jaeger/cmd/collector/app/server.serveHTTP.func1
        /Users/limistah/Work/oss/jaeger/cmd/collector/app/server/http.go:111
--- FAIL: TestSpanCollectorHTTPS (0.02s)
    --- FAIL: TestSpanCollectorHTTPS/should_fail_with_TLS_client_to_untrusted_TLS_server (0.02s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x40 pc=0x104e6664c]